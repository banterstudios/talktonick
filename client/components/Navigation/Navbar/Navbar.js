import React, { Component } from 'react'
import glamorous from 'glamorous'
import PropTypes from 'prop-types'

const NavbarWrapper = glamorous.nav(({ theme: { headerHeight }, direction }) => ({
  position: 'relative',
  display: 'flex',
  flexDirection: 'row',
  flexWrap: 'nowrap',
  height: headerHeight,
  width: '100%',
  justifyContent: (direction === 'left' ? 'flex-start' : direction === 'center' ? 'center' : 'flex-end')
}))

export default class Navbar extends Component {
  static propTypes = {
    direction: PropTypes.string
  }

  render () {
    const { children } = this.props

    return (
      <NavbarWrapper role='navigation'>
        { children }
      </NavbarWrapper>
    )
  }
}
