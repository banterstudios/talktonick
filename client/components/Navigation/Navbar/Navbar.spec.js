import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Navbar from './Navbar'

describe('components/Navigation/Navbar', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = mount(<Navbar />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
