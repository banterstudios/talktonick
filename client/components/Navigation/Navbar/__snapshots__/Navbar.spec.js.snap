// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation/Navbar @renders should render correctly 1`] = `
<Navbar>
  <glamorous(nav)
    role="navigation"
  >
    <nav
      className="css-1b5l5ja"
      role="navigation"
    />
  </glamorous(nav)>
</Navbar>
`;
